tests/steamship_tests/plugin/unit/test_capabilities.py

```diff
@@ -0,0 +1,114 @@
+from typing import List
+
+import pytest
+
+from steamship import Block, Tag
+from steamship.data import TagKind
+from steamship.data.tags.tag_constants import ChatTag
+from steamship.plugin.capabilities import (
+    Capability,
+    CapabilityImpl,
+    CapabilityPluginRequest,
+    CapabilityPluginResponse,
+    RequestedCapabilities,
+    RequestLevel,
+    UnsupportedCapabilityError,
+)
+
+
+class TestCapability(CapabilityImpl):
+    NAME = "steamship.test_capability"
+
+
+class AnotherTestCapability(CapabilityImpl):
+    NAME = "steamship.test_capability_2"
+
+
+@pytest.mark.parametrize(
+    ("request_level", "support_level", "expected_support_valid", "expected_none_response"),
+    [
+        (RequestLevel.NATIVE, RequestLevel.NATIVE, True, False),
+        (RequestLevel.NATIVE, RequestLevel.BEST_EFFORT, False, True),
+        (RequestLevel.NATIVE, None, False, True),
+        (RequestLevel.BEST_EFFORT, RequestLevel.NATIVE, True, False),
+        (RequestLevel.BEST_EFFORT, RequestLevel.BEST_EFFORT, True, False),
+        (RequestLevel.BEST_EFFORT, None, False, True),
+        (RequestLevel.OPTIONAL, RequestLevel.NATIVE, True, False),
+        (RequestLevel.OPTIONAL, RequestLevel.BEST_EFFORT, True, False),
+        (RequestLevel.OPTIONAL, None, True, True),
+    ],
+)
+def test_is_plugin_support_valid(
+    request_level, support_level, expected_support_valid, expected_none_response
+):
+    request = TestCapability(request_level=request_level)
+    is_valid, response = request.is_plugin_support_valid(support_level)
+    assert is_valid == expected_support_valid
+    if expected_none_response:
+        assert response is None
+    else:
+        assert response is not None
+        assert response.fulfilled_at == support_level
+
+
+def test_capability_plugin_request_block_roundtrips():
+    original = CapabilityPluginRequest(
+        requested_capabilities=[TestCapability(request_level=RequestLevel.NATIVE)]
+    )
+    block = original.to_block()
+    roundtripped = CapabilityPluginRequest.from_block(block)
+    assert original == roundtripped
+
+
+def test_capability_plugin_response_block_roundtrips():
+    original = CapabilityPluginResponse(
+        capability_responses=[Capability.Response(fulfilled_at=RequestLevel.NATIVE)]
+    )
+    block = original.to_block()
+    roundtripped = CapabilityPluginResponse.from_block(block)
+    assert original == roundtripped
+
+
+def _make_input_blocks() -> List[Block]:
+    return [
+        Block(
+            text="Can you make me an image of a whale?",
+            tags=[Tag(kind=TagKind.CHAT, name=ChatTag.HISTORY)],
```

Comment:
FWIW, so far, we only use `ChatTag.HISTORY` on the `File`-level tags for the `ChatHistory` file. There are several reasons for this, including that a message is only `HISTORY` based on other context. For instance, at first, it is just a `MESSAGE`.